datasets/waveform_21/metadata.yaml

```diff
@@ -1,19 +1,28 @@
 #Generated automatically by pmlb/write_metadata.py
 dataset: waveform_21
-description: None yet. See our contributing guide to help us add one.
-source: None yet. See our contributing guide to help us add one.
-publication: None yet. See our contributing guide to help us add one.
+description: >
+  Three classes of waves, twenty one attributes.
+  Each class is generated from a combination of 2 of 3 "base" waves with added noise.
+  All attributes include noise.
+source: https://www.openml.org/d/4551
+publication: >
+  https://archive.ics.uci.edu/ml/datasets/Waveform+Database+Generator+%28Version+1%29
+  Breiman, L., Friedman, J.H., Olshen, R.A., and Stone, C.J. (1984). 
+  Classification and Regression Trees. 
+  Wadsworth International, pp 49-55, 169.
+  doi: 10.1201/9781315139470
 task: classification
 target:
   type: discrete
-  description: None yet. See our contributing guide to help us add one.
-  code: None yet. See our contributing guide to help us add one.
+  description: One of the three classes of wave; determined by the 21 attributes.
+  code: >
+    'V1'='X00', 'V2'='X01', 'V3'='X02', 'V4'='X03', 'V5'='X04', 'V6='X05','V7'='X06', 'V8'='X07', 'V9'='X08', 'V10'='X09',
```

Comment:
This field is supposed to be description for different values of target (0, 1, 2). I don't think the mapping of feature names are too important because they're all simulated variables, but you can move this to the dataset general description (line 3) if you'd like.

Reply:
I agree - I think a general description of the pattern should be fine.